introduction-to-datasets.md

```diff
@@ -0,0 +1,119 @@
+[api-schema-from-data]: {{site.baseurl}}/Inferring-a-Schema-from-an-Avro-Data-File.html
+[cli-schema-from-class]: {{site.baseurl}}/cli-reference.html#obj-schema
+[cli-schema-from-csv]: {{site.baseurl}}/cli-reference.html#csv-schema
+
+## Partitions
+
+Partitions define logical categories for data storage. For example, you might most often retrieve your data using time-based queries. You can define a partitioning strategy by year, month, day, and hour. When you look for data from January 8, 2015, your search engine only has to look in the data partition `/year=2015/month=1/day=8`. By using partitions that correspond to your most common queries, your data searches run more quickly.
+
+Partitioning is optional, because there are times when partitioning is not the most efficient solution. However, you should always consider partitioning as a best practice when planning your dataset.
+
+When using the CLI, you define your partition strategy in [JSON format][json-format] and apply it when you create your dataset. See [Partitioned Datasets][partition-strategy].
+
+When using the API, you define your dataset in the `DatasetDescriptor.Builder`. See [Partition Builder][partition-builder].
+
+For example, you can use the CLI command `partition-config` to define a partition strategy using name:value pairs to specify that the dataset should be partitioned on a timestamp (`ts`) field by year, month, and day.
+
+```
+user@work:~$ kite-dataset partition-config ts:year ts:month ts:day -s rating.avsc
+```
+The result is a partition definition in JSON format. You can use the `-o` option to out put the result and store it in a .json file.
+
+```JSON
+[ {
+  "name" : "year",
+  "source" : "ts",
+  "type" : "year"
+}, {
+  "name" : "month",
+  "source" : "ts",
+  "type" : "month"
+}, {
+  "name" : "day",
+  "source" : "ts",
+  "type" : "day"
+} ]
+```
+
+[json-format]: {{site.baseurl}}/Partition-Strategy-Format.html
+[partition-strategy]: {{site.baseurl}}/Partitioned-Datasets.html#partition-strategies
+[partition-builder]: {{site.baseurl}}/API-Overview.html#partition-strategy
+
+## Configuration Options
+
+While you need only a URI and schema to create a dataset, you can enhance the performance of your dataset with additional configuration.
+
+### Column Mapping
+
+Column mapping allows you to configure how your records should be stored in HBase for maximum performance and efficiency. You define column mapping in JSON format in a data-centric way. Kite stores and retrieves the data correctly. 
+
+See [Column Mapping][column-mapping]. See also the [CLI mapping-config command][cli-mapping-config].
+
+[column-mapping]: {{site.baseurl}}/Column-Mapping.html
+[cli-mapping-config]: {{site.baseurl}}/cli-reference.html#mapping-config
+
+### Properties
+
+You can add custom settings to dataset properties at creation time using the `--set` option. For example, in the CLI you might create the dataset _users_ using a cache size of 20 (rather than the default cache size of 10):
```

Comment:
The part about the `--set` option is CLI-specific and should be part of the example, but not part of the paragraph explaining what properties do.

This should also briefly explain what the writer cache size controls, which is the number of files that are kept open at one time. One file is used for each partition that you write to.